src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +36,534 @@
+            control_config_obj = NodeConfig(controller)
+            control_config_obj.ssh_connect()
+            control_config_obj.update_filter()
+    except Exception as error:
+        message = "Exception {}: {}".format(type(error).__name__, str(error))
+        logger.error("{}".format(message))
+
+
+def create_custom_flavors(overcloud_name,
+                          enable_hugepage,
+                          enable_numa, hugepage_size,
+                          hpg_flavor_names_list,
+                          numa_flavor_names_list):
+    try:
+        if enable_hugepage or enable_numa:
+            logger.info("Create custom flavors")
+
+            # Get the overcloud details
+            os_auth_url, os_tenant_name, os_username, os_password = \
+                CredentialHelper.get_overcloud_creds()
+
+            kwargs = {'username': os_username,
+                      'password': os_password,
+                      'auth_url': os_auth_url,
+                      'project_id': os_tenant_name}
+
+            # Create nova client object
+            nova = nova_client.Client(2, **kwargs)
+
+            # Create the ConfigOvercloud object
+            config_oc_obj = ConfigOvercloud(nova)
+            # Create falvors for HugePage feature and set its metadata
+            if enable_hugepage:
+                metadata = ConfigOvercloud.get_hp_flavor_meta(hugepage_size)
+                for flavor_name in hpg_flavor_names_list:
+                    flavor = config_oc_obj.create_flavor(flavor_name)
+                    config_oc_obj.set_flavor_metadata(flavor, metadata)
+
+            # Create falvors for NUMA feature and set its metadata
+            if enable_numa:
+                flavor_meta = {'aggregate_instance_extra_specs:pinned': 'True',
+                               'hw:cpu_policy': 'dedicated',
+                               'hw:cpu_thread_policy': 'require'}
+                for flavor_name in numa_flavor_names_list:
+                    flavor = config_oc_obj.create_flavor(flavor_name)
+                    config_oc_obj.set_flavor_metadata(flavor, flavor_meta)
+    except Exception as error:
+        message = "Exception {}: {}".format(type(error).__name__, str(error))
+        logger.error("{}".format(message))
+
+
+def get_dell_compute_nodes_uuids():
+    logger.info("Getting dell compute node uuids")
+    # Get the undercloud details
+    os_auth_url, os_tenant_name, os_username, os_password = \
+        CredentialHelper.get_undercloud_creds()
+
+    kwargs = {'username': os_username,
+              'password': os_password,
+              'auth_url': os_auth_url,
+              'project_id': os_tenant_name}
+
+    # Create nova client object
+    nova = nova_client.Client(2, **kwargs)
+
+    # Create the ConfigOvercloud object
+    config_oc_obj = ConfigOvercloud(nova)
+    dell_compute_nodes_uuid = config_oc_obj.get_dell_compute_nodes_uuid()
+    logger.info("{}".format(dell_compute_nodes_uuid))
```

Comment:
The above logging statement should probably be at the debug level, and should include some indication as to what is being logged like "dell_compute_nodes_uuid: {}".format ...

Reply:
Yeah, makes sense. Changed.